.github/workflows/doxygen.yml

```diff
@@ -0,0 +1,30 @@
+name: Doxygen
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  Build and Publish Documentation:
+    name: Build and Publish Documentation
+    runs-on: ubuntu-latest
+    container:
+      image: novelrt/novelrt-build:latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Generate Documentation
+        run: scripts/cibuild.sh --documentation -DDOXYGEN_OUTPUT_DIR=$GITHUB_WORKSPACE/novelrt/docs
+
+      - name: Commit Documentation Updates
+        if: ${{ success() }}
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git checkout gh-pages
+          git add ./docs/*
+          git commit -m "Updated documentation - $GITHUB_SHA"
+          git push
```

Comment:
I feel like this should be creating a PR to the pages repo and applying auto-merge once CI/sign-off completes

Reply:
Would you be able to assist with setting up auto-merge? This is something I'm very unfamiliar with 😅

Also, so I understand the flow desired:
Checkout to new feature branch
Add the docs folder that was generated
Commit to the feature branch
Push (to remote)
Request PR between the feature branch and the `gh-pages` branch

Does this seem correct?

Reply:
Tried to implement this (GH doesn't provide a stock "Create Pull Request" action so I had to use one from the marketplace instead.)